qiita_client/qiita_client.py

```diff
@@ -556,3 +560,106 @@ def complete_job(self, job_id, success, error_msg=None,
                                              artifacts_info=artifacts_info))
         # Create the URL where we have to post the results
         self.post("/qiita_db/jobs/%s/complete/" % job_id, data=json_payload)
+
+    def artifact_and_preparation_files(self, artifact_id,
+                                       ignore_small_files=True):
+        """Gets the artifact and preparation files from a given artifact_id
+
+        Parameters
+        ----------
+        artifact_id : int
+            The artifact id
+        ignore_small_files : bool
+            Whether to ignore small files or retrieve all of them (only applies
+            to per_sample_FASTQ artifacts)
+
+        Returns
+        -------
+
+        Raises
+        ------
+        RuntimeError
+            - If the artifact belongs to an analysis
+
+        """
+        artifact_info = self.get("/qiita_db/artifacts/%s/" % artifact_id)
+
+        if artifact_info['analysis'] is not None:
+            raise RuntimeError(
+                f'Artifact {artifact_id} is an analysis artifact, this method '
+                'is meant to work with artifacts linked to a preparation.')
+
+        prep_info = self.get('/qiita_db/prep_template/%s/'
+                             % artifact_info['prep_information'][0])
+        prep_info = pd.read_csv(prep_info['prep-file'], sep='\t', dtype=str)
+        if artifact_info['type'] == 'per_sample_FASTQ':
+            files, prep_info = self._process_files_per_sample_fastq(
+                artifact_info['files'], prep_info, ignore_small_files)
+        else:
+            files = {k: [vv['filepath'] for vv in v]
+                     for k, v in artifact_info['files'].items()}
+
+        return files, prep_info
+
+    def _process_files_per_sample_fastq(self, files, prep_info,
+                                        ignore_small_files):
+        "helper function to process per_sample_FASTQ artifacts and their preps"
+
+        fwds = sorted(files['raw_forward_seqs'], key=lambda x: x['filepath'])
+        revs = []
+        if 'raw_reverse_seqs' in files:
+            revs = sorted(
+                files['raw_reverse_seqs'], key=lambda x: x['filepath'])
+            if len(fwds) != len(revs):
+                raise ValueError(f'The fwd ({len(fwds)}) and rev ({len(revs)})'
+                                 ' files should be of the same length')
+
+        run_prefixes = prep_info['run_prefix'].to_dict()
+
+        # make parirings
+        sample_names = dict()
+        used_prefixes = []
+        for i, (fwd, rev) in enumerate(zip_longest(fwds, revs)):
+            fwd_fn = basename(fwd['filepath'])
+            file_smaller_than_min = fwd['size'] < MIN_FILEPATH_SIZE
+
```

Comment:
The name MIN_FILEPATH_SIZE makes me think of the minimum length a file_path has to be in order to be valid. Really this value is meant to be the minimum size for the file named 'filepath'. Maybe BLANK_FILE_THRESHOLD or something similar might be more descriptive?